wicket/src/ui/panes/overview.rs

```diff
@@ -439,3 +444,429 @@ impl Control for InventoryView {
+            } => {
+                spans.push(label.into());
+                spans.push(
+                    vec![
+                        bullet(),
+                        Span::styled("Identity: ", label_style),
+                        Span::styled(format!("{id:?}"), ok_style),
+                    ]
+                    .into(),
+                );
+                spans.push(
+                    vec![
+                        bullet(),
+                        Span::styled("Power: ", label_style),
+                        Span::styled(yes_no(*power), dyn_style(*power)),
+                    ]
+                    .into(),
+                );
+                if *flt_a2 || *flt_a3 || *flt_rot || *flt_sp {
+                    let mut faults =
+                        vec![bullet(), Span::styled("Faults:", label_style)];
+                    if *flt_a2 {
+                        faults.push(Span::styled(" A2", bad_style));
+                    }
+                    if *flt_a3 {
+                        faults.push(Span::styled(" A3", bad_style));
+                    }
+                    if *flt_rot {
+                        faults.push(Span::styled(" RoT", bad_style));
+                    }
+                    if *flt_sp {
+                        faults.push(Span::styled(" SP", bad_style));
+                    }
+                    spans.push(faults.into());
+                } else {
+                    spans.push(
+                        vec![
+                            bullet(),
+                            Span::styled("Faults: ", label_style),
+                            Span::styled("None", ok_style),
+                        ]
+                        .into(),
+                    );
+                }
+                spans.push(
+                    vec![
+                        bullet(),
+                        Span::styled("Controller 0 detected: ", label_style),
+                        Span::styled(
+                            yes_no(*ctrl_detect_0),
+                            dyn_style(*ctrl_detect_0),
+                        ),
+                    ]
+                    .into(),
+                );
+                spans.push(
+                    vec![
+                        bullet(),
+                        Span::styled("Controller 1 detected: ", label_style),
+                        Span::styled(
+                            yes_no(*ctrl_detect_1),
+                            dyn_style(*ctrl_detect_1),
+                        ),
+                    ]
+                    .into(),
+                );
+            }
+        }
+    } else {
+        label.push(Span::styled("Not available", bad_style));
+        spans.push(label.into());
+    }
+
+    // blank line separator
+    spans.push(Spans::default());
+
+    // Helper closure for appending caboose details (used for both SP and RoT
+    // below).
+    let append_caboose =
+        |spans: &mut Vec<Spans>, caboose: &SpComponentCaboose| {
```

Comment:
This can be a `fn append_caboose` I think since it doesn't capture anything in the environment.

Reply:
It captured a few small helpers, but those were easy enough to repeat or pass in; moved in 075fb020